store-implementation/federated-store/src/main/java/uk/gov/gchq/gaffer/federatedstore/FederatedStore.java

```diff
@@ -102,9 +111,21 @@
  * @see Graph
  */
 public class FederatedStore extends Store {
+    private static final Logger LOGGER = LoggerFactory.getLogger(Store.class);
+    private static final String FEDERATED_STORE_PROCESSED = "FederatedStore.processed.";
     private FederatedGraphStorage graphStorage = new FederatedGraphStorage();
     private Set<String> customPropertiesAuths;
     private Boolean isPublicAccessAllowed = Boolean.valueOf(IS_PUBLIC_ACCESS_ALLOWED_DEFAULT);
+    private static final List<Integer> ALL_IDS = new ArrayList<>();
+    private final int id;
+
+    public FederatedStore() {
+        Integer i = null;
+        while (isNull(i) || ALL_IDS.contains(i)) {
+            i = new Random().nextInt();
+        }
+        ALL_IDS.add(id = i);
```

Comment:
Does this mean that two FederatedStores in seperate JVMs could have the same id?

Reply:
Yes, but collision is chance is very low. This is to mitigate against infinite loops not guarantee it doesn't ever happen. Next time one of the JVM is turned on and off again it should be okay.